src/Components/Users/Users.tsx

```diff
@@ -84,12 +84,14 @@ export default function Users() {
   }
 
   useEffect(() => {
-    getAllUsers()
+    const timerId = setTimeout(()=>{
+      getAllUsers()
+    },500)
+    return () => clearTimeout(timerId)
   }, [])
 
   // Search by name
-  const getNameValue = (input) => {
```

Comment:
We should implement debouncing for the API call within this function, as it's invoked on every input change. In the existing setup, we are merely introducing a delay to the initial API call to `getAllUsers` during the first render.

Also, keep in mind the availability of the network tap for testing. It's a valuable tool to assess the functionality and effectiveness of your implementation.